### ISSUE TITLE
Uninitialized len in handle_fm7_socket_data

### ISSUE THREAD

**Issue by robominds** [MatthewCash/libforza#2]
When compiling libforza on a Raspberry Pi 4 with gcc 8.3.0, recvfrom in handle_fm7_socket was returning -1 in msg_len indicating an error condition. Adding a perror call after recvfrom I was getting an "invalid argument" error. I found that by initializing len to either zero or sizeof(struct sockaddr) then recvfrom would work as expected.

I compiled libforza on Ubuntu/gcc 9.3.0 and the above issues were not encountered and recvfrom worked as expected.

From https://linux.die.net/man/2/recvfrom, "The argument addrlen is a value-result argument, which the caller should initialize before the call to the size of the buffer associated with src_addr".

For robustness, I suggest initializing len to sizeof(struct sockaddr).

**Comment by MatthewCash** [comment]
I have only tested on Ubuntu 20.04 (GCC 9.3.0-17) and NixOS Unstable (GCC 10.3.0) so I did not encounter this issue, but it is certainly an error in the library's current method of error handling and use of `recvfrom`. Thank you for pointing this out, I definitely want this to be compatible with as many Linux platforms as possible, so fixing this will be important.

I am planning on making some improvements regarding robustness and error handling, but haven't thought about it recently. I'll work on this hopefully tomorrow (2022/2/1) or the day after. 

Also just a clarification, this is a runtime error, not a compile time error, correct?

**Comment by robominds** [comment]
Correct, it is a runtime error.

**Comment by robominds** [comment]
Just to be clear, I do not think this is an error handling change. This seems to be differences in how 'recvfrom' is implemented on different platforms. On the RPi, the 'len' argument needs to be initialized while on Ubuntu it does not.

The change that fixes the recvfrom call in 'handle_fm7_socket' is:
```
-    socklen_t len;
+    socklen_t len = sizeof(struct sockaddr);
```

**Comment by MatthewCash** [comment]
I understand, I have been working on improving error handling and my working changes affect parts of the `handle_xxx_socket_data` function to make sure `src_addr` and `addrlen` are setup/used properly.

**Comment by robominds** [comment]
The fix in #5 worked for me under Ubuntu 20.04 and RPi Raspbian buster.
